nupic/data/record_stream.py

```diff
@@ -25,182 +25,272 @@
-    timestamp index - this is the timestamp divided by the aggregation period. 
-    
-    
+    timestamp index - this is the timestamp divided by the aggregation period.
+
+
     Parameters:
     ------------------------------------------------------------------------
     recordTS:  datetime instance
-    retval:    record timestamp index, or None if no aggregation period 
+    retval:    record timestamp index, or None if no aggregation period
     """
-    
-    aggPeriod = self.getAggregationMonthsAndSeconds()
-    if aggPeriod is None:
+
+    if self._aggregationPeriod is None:
       return None
-    
-    # Base record index on number of elapsed months if aggregation is in 
+
+    # Base record index on number of elapsed months if aggregation is in
     #  months
-    if aggPeriod['months'] > 0:
-      assert aggPeriod['seconds'] == 0
-      result = \
-        int((recordTS.year * 12 + (recordTS.month-1)) / aggPeriod['months'])
-        
+    if self._aggregationPeriod['months'] > 0:
+      assert self._aggregationPeriod['seconds'] == 0
+      result = int(
+        (recordTS.year * 12 + (recordTS.month-1)) /
+        self._aggregationPeriod['months'])
+
     # Base record index on elapsed seconds
-    elif aggPeriod['seconds'] > 0:
+    elif self._aggregationPeriod['seconds'] > 0:
       delta = recordTS - datetime.datetime(year=1, month=1, day=1)
       deltaSecs = delta.days * 24 * 60 * 60   \
                 + delta.seconds               \
                 + delta.microseconds / 1000000.0
-      result = int(deltaSecs / aggPeriod['seconds'])
-    
+      result = int(deltaSecs / self._aggregationPeriod['seconds'])
+
     else:
       result = None
-      
+
     return result
 
 
-  def getAggregationMonthsAndSeconds(self):
-    """ Returns the aggregation period of the record stream as a dict 
-    containing 'months' and 'seconds'. The months is always an integer and
-    seconds is a floating point. Only one is allowed to be non-zero.  
-    
-    If there is no aggregation associated with the stream, returns None. 
-    
-    Typically, a raw file or hbase stream will NOT have any aggregation info,
-    but subclasses of RecordStreamIFace, like StreamReader, will and will
-    return the aggregation period from this call. This call is used by the
-    getNextRecordDict() method to assign a record number to a record given
-    its timestamp and the aggregation interval
-    
-    Parameters:
-    ------------------------------------------------------------------------
-    retval: aggregationPeriod (as a dict) or None  
-              'months': number of months in aggregation period
-              'seconds': number of seconds in aggregation period (as a float)
+
+class RecordStreamIface(object):
+  """This is the interface for the record input/output storage classes."""
+
+  __metaclass__ = ABCMeta
+
+
+  def __init__(self, fields, aggregationPeriod=None):
```

Comment:
Note that the args in the `RecordStreamIface` constitute a change in its interface versus `def __init__(self)`. The legacy code was a bit backwards in that it depended on abstract methods to query this **static** information on demand each time it was needed (and it needed these values) instead of simply getting it via constructor as is typical.